### DISCUSSION TITLE
How do you replace / port storage backend as rocksdb instead of lmdb?

### DISCUSSION THREAD

**Discussion by hiqsociety** [meilisearch/product#158] [Feedback & Feature Proposal]
How do you replace / port storage backend as rocksdb instead of lmdb? or rocksdb compatible solution or redis?

**Comment by curquiza** [comment]
Hello @hiqsociety! This is impossible at the moment, MeiliSearch is written to work with LMDB and is not KV store agnostic.

**Comment by gmourier** [comment]
Hi @hiqsociety, for which use-cases/benefits would you change LMDB to RocksDB? I'm curious to know more. I imagine for faster writing needs.

The core team is currently working on optimizing the indexing speed and this should be improved by v0.22.

**Reply by hiqsociety** [reply]
yes definitely faster write speed. possible to use only rocksdb version 5? it's write speed is much much better

**Comment by gmourier** [comment]
Hey 👋

I'm locking that discussion because Meilisearch won't use rocksdb.

Write speed has been heavily worked on in the last versions. Please let us know [here](https://github.com/meilisearch/meilisearch/discussions/2131) if you are still encountering indexing speed issues with the last version.

Thanks!
